gcapi/create_strategies.py

```diff
@@ -10,27 +11,93 @@
 if TYPE_CHECKING:
     from gcapi.client import Client
 
-from gcapi.exceptions import ObjectNotFound
 from gcapi.models import (
     Algorithm,
     ComponentInterface,
     ComponentInterfaceValuePostRequest,
     HyperlinkedComponentInterfaceValue,
-    HyperlinkedImage,
     UserUpload,
 )
-from gcapi.typing import FileSource, SocketValueSetDescription
 
 
 class TooManyFiles(ValueError):
     pass
 
 
-Empty = object()
+class UnsetType:
+    pass
+
+
+Unset = UnsetType()
+
+
+@dataclass
+class SocketValueSpec:
+    """
+    Specification for a socket value which provides exactly one source
+    in order to create the SocketValue.
+
+    Args:
+        socket_slug: The slug of the socket (required)
+
+        value: Direct source value to set (can be None)
+        files: List of source file paths for file/image uploads
+        existing_image_api_url: An API URL of an existing image to reuse
+        existing_socket_value: Existing socket value that can be reused
+    """
+
+    socket_slug: str
+
+    value: Any = Unset
+    files: list[str | Path] | UnsetType = Unset
+
+    existing_image_api_url: str | UnsetType = Unset
+    existing_socket_value: HyperlinkedComponentInterfaceValue | UnsetType = (
+        Unset
+    )
+
+    def __post_init__(self):
+        """Validate that the specification is correct."""
+        # Get all fields except socket_slug
+        sources = []
+
+        for field_name, field_value in self.__dict__.items():
+            if field_name == "socket_slug":
+                continue
+            if field_value is not Unset:
+                sources.append(field_name)
+
+        if len(sources) > 1:
+            raise ValueError(
+                f"Only one source can be specified, but got: {', '.join(sources)}. "
+                f"Please specify only one of the available source fields."
+            )
+
+        if len(sources) == 0:
+            raise ValueError(
+                "At least one source must be specified. "
+                "Please provide one of the available source fields."
+            )
+
+    def get_set_field_name(self) -> str:
+        for field_name, field_value in self.__dict__.items():
+            if field_name == "socket_slug":
+                continue
+            if field_value is not Unset:
+                return field_name
+        raise RuntimeError("No source field is set, this should not happen.")
+
+    def __repr__(self) -> str:
+        set_field_name = self.get_set_field_name()
+        set_field_value = getattr(self, set_field_name)
+        return (
+            f"{self.__class__.__name__}({self.socket_slug=}, "
```

Comment:
This results in `SocketValueSpec(self.socket_slug='foo', ...)` which is not valid. 
```suggestion
            f"{self.__class__.__name__}(socket_slug={self.socket_slug}, "
```